native/cocos/renderer/gfx-base/GFXDevice.cpp

```diff
@@ -105,9 +107,11 @@ void Device::destroySurface(void *windowHandle) {
 }
 
 void Device::createSurface(void *windowHandle) {
+    auto windowId = static_cast<uint32_t>(reinterpret_cast<uintptr_t>(windowHandle));
     for (const auto &swapchain : _swapchains) {
-        if (!swapchain->getWindowHandle()) {
-            swapchain->createSurface(windowHandle);
+        if (swapchain->getWindowId() == windowId) {
+            auto *window = CC_GET_SYSTEM_WINDOW(windowId);
+            swapchain->createSurface(reinterpret_cast<void *>(window->getWindowHandle()));
```

Comment:
Here the dependency on platform might cause webasm fail to compile. Consider removing `ISystemWindow.h` and `ISystemWindowManager.h`.
Did we changed the `void* windowHandle` to `uint32_t windowId`?
If it is the case, we might need to change the signature of `gfx::Device::createSurface`.

Reply:
I'll check it.